user-guide/Advanced_Functionality/DataMiner_Agents/Upgrading_a_DMA/Preparing_to_upgrade_a_DataMiner_Agent.md

```diff
@@ -33,14 +35,93 @@ To upload an upgrade package:
+
+### Best practices when upgrading across major versions
+
+If you are not upgrading to the DataMiner version that immediately follows your current version, always follow this general guideline:
+
+**Upgrade to the next major DataMiner version(s) before upgrading to your preferred version**.
+
+> [!IMPORTANT]
+>
+> - There is one exception to this rule: if you are upgrading from a version prior to DataMiner 10.1.x to 10.2.x or higher, **skip DataMiner 10.1** and proceed directly to the subsequent major version instead (i.e. DataMiner 10.2).
+> - If you are upgrading from DataMiner 9.6.0/9.6.x, upgrade to [DataMiner 10.0.0 [CU19]](https://community.dataminer.services/download/dataminer-main-release-10-0-0-0-11025-cu19/) as the next major version.
+
+> [!TIP]
+> See [Example upgrade path](#example-upgrade-path).
+
+#### Upgrade prerequisites
+
+If you are about to upgrade across major versions, before proceeding with the upgrade, **ensure the following requirements are met**:
+
+- Depending on the target DataMiner version, make sure the corresponding Microsoft .NET and ASP.NET Core packages are installed:
+
+  | DataMiner version | Microsoft .NET | ASP.NET Core |
+  |--|--|--|
+  | DataMiner 10.3.0 [CU12]/10.4.0/10.4.3 and higher | [Microsoft .NET 4.8](https://go.microsoft.com/fwlink/?linkid=2088631) | [ASP.NET Core 8.0](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-aspnetcore-8.0.1-windows-hosting-bundle-installer) |
+  | DataMiner 10.3.9/10.4.1/10.4.2 | [Microsoft .NET 4.8](https://go.microsoft.com/fwlink/?linkid=2088631) | [ASP.NET Core 6.0.13](https://download.visualstudio.microsoft.com/download/pr/0cb3c095-c4f4-4d55-929b-3b4888a7b5f1/4156664d6bfcb46b63916a8cd43f8305/dotnet-hosting-6.0.13-win.exe) |
+  | DataMiner 10.3.0 main release versions from 10.3.0 [CU3] onwards<br/>DataMiner 10.3.3 to 10.3.8 | [Microsoft .NET 4.8](https://go.microsoft.com/fwlink/?linkid=2088631) | [ASP.NET Core 5.0.11](https://download.visualstudio.microsoft.com/download/pr/df452763-4b7d-490a-bc03-bd1003d3ff4c/665ee1786528809f33e791558b69cf51/dotnet-hosting-5.0.11-win.exe)<br/>[ASP.NET Core 6.0.13](https://download.visualstudio.microsoft.com/download/pr/0cb3c095-c4f4-4d55-929b-3b4888a7b5f1/4156664d6bfcb46b63916a8cd43f8305/dotnet-hosting-6.0.13-win.exe) |
+  | DataMiner 10.3.0 [CU0] up to [CU2]<br/>DataMiner 10.2.0/10.2.x | [Microsoft .NET 4.8](https://go.microsoft.com/fwlink/?linkid=2088631) | [ASP.NET Core 5.0.11](https://download.visualstudio.microsoft.com/download/pr/df452763-4b7d-490a-bc03-bd1003d3ff4c/665ee1786528809f33e791558b69cf51/dotnet-hosting-5.0.11-win.exe) |
+
+  > [!IMPORTANT]
+  > If you do not install these packages before executing the upgrade, the DataMiner installer will install them for you, but this will require a reboot during the upgrade. As this can make following the upgrade process more difficult, we recommend installing these packages in advance.
```

Comment:
This does not apply for ASP .NET 8.0; if this is not installed, you won't be able to run the upgrade at all.
We could either detail this in this note, or leave out the note completely (because there will probably be more and more exceptions to it in the long run). What do you think?

Reply:
I believe we could indeed leave out the note.